### PR TITLE
Support multi-instance terminate instructions

### LINKED ISSUE #9701: Modify a process instance by terminating an element instance of a multi-instance activity
## Description

We want to support the termination of an element instance that belongs to a multi-instance. As a result, all element instances inside the multi-instance and the multi-instance itself should be terminated. 

If an element instance inside the multi-instance has also child instances (i.e. a nested subprocess) then these child instances should be terminated as well (recursively). 

We should produce the same sequence of events as a regular termination (i.e. `multi-instance-body: terminating`, `element: terminating`, `element: terminated`, `multi-instance-body: terminated`). 

For example:
image

* terminate the multi-instance (according to #9690)
* terminate all active element instances inside this multi-instance
* propagate the termination of the multi-instance to its flow scope (according to #9691)  

Blocked by 
- #9691

### PR DESCRIPTION
**Pull request by korthout**
## Description

When a terminate instruction (of a process instance modification) refers to a multi-instance element, all element instances inside the multi-instance and the multi-instance itself should be terminated.

## Related issues

closes #9701

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [x] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java [hunk 1]
```diff
@@ -760,6 +760,157 @@ public void shouldTerminateEmbeddedSubProcess() {
...
+  public void shouldTerminateParallelMultiInstanceElement() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .parallelGateway()
+                .userTask("A")
+                .moveToLastGateway()
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java:773]: 🙃 I don't think we need this gateway and user task A

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java [hunk 2]
```diff
@@ -760,6 +760,157 @@ public void shouldTerminateEmbeddedSubProcess() {
...
+  public void shouldTerminateSequentialMultiInstanceElement() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .parallelGateway()
+                .userTask("A")
+                .moveToLastGateway()
```
Comment (remcowesterhoud) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java:854]: 🙃 Idem

**Review by remcowesterhoud — APPROVED** [review]
LGTM! 🚀 

Let the battle of the merge conflicts commence 😄

File: engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java (hunk 1 above)
Comment (remcowesterhoud) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java:773]: Does this matter? It gets terminated after processing the modification, after the multi-instance has been terminated.

Comment (remcowesterhoud) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java:773]: You sequence could become:
```java
        .containsSequence(
             tuple(BpmnElementType.MULTI_INSTANCE_BODY, ProcessInstanceIntent.ELEMENT_TERMINATING),
             tuple(BpmnElementType.SUB_PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATING),
             tuple(BpmnElementType.USER_TASK, ProcessInstanceIntent.ELEMENT_TERMINATING),
             tuple(BpmnElementType.USER_TASK, ProcessInstanceIntent.ELEMENT_TERMINATED),
             tuple(BpmnElementType.SUB_PROCESS, ProcessInstanceIntent.ELEMENT_TERMINATED),
             tuple(BpmnElementType.MULTI_INSTANCE_BODY, ProcessInstanceIntent.ELEMENT_TERMINATED),
            tuple(BpmnElementType.PROCESS, PROCESS_ID, ProcessInstanceIntent.ELEMENT_TERMINATING),
            tuple(BpmnElementType.PROCESS, PROCESS_ID, ProcessInstanceIntent.ELEMENT_TERMINATED));
```

Comment (korthout) [reply engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceTerminationTest.java]: Came out pretty nice, I like how I could also simplify the limits this way

**Pull request merged**